src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -88,15 +90,17 @@
     protected List<String> invites = null;
 
     public ReceivedMessage(
-        long id, MessageChannel channel, MessageType type, Message referencedMessage,
+        long id, MessageChannel channel, MessageType type, MessageReference messageReference,
         boolean fromWebhook, boolean mentionsEveryone, TLongSet mentionedUsers, TLongSet mentionedRoles, boolean tts, boolean pinned,
         String content, String nonce, User author, Member member, MessageActivity activity, OffsetDateTime editTime,
         List<MessageReaction> reactions, List<Attachment> attachments, List<MessageEmbed> embeds, List<MessageSticker> stickers, List<ActionRow> components, int flags)
     {
         super(content, nonce, tts);
         this.id = id;
         this.channel = channel;
-        this.referencedMessage = referencedMessage;
+        this.messageReference = messageReference;
+        this.referencedMessage = (messageReference != null) ? messageReference.getReferencedMessage() : null;
```

Comment:
Why not just do `messageReference.getReferenedMessage()` where needed in the code instead of storing it here?